src/lib/snark_worker/prod.ml

```diff
@@ -92,12 +92,64 @@ module Inputs = struct
               | Work.Single.Spec.Transition (input, (w : Transaction_witness.t))
                 ->
                   process (fun () ->
-                      match w with
-                      | Parties_segment witness ->
-                          Deferred.Or_error.try_with ~here:[%here] (fun () ->
-                              M.of_parties_segment_exn
-                                ~statement:{ input with sok_digest } ~witness)
-                      | Non_parties w ->
+                      match w.transaction with
+                      | Command (Parties parties) ->
+                          let witnesses_specs_stmts =
+                            Transaction_snark.parties_witnesses_exn
```

Comment:
It seems a pity to have to recompute this when we've already evaluated the transaction once in the staged ledger..

Reply:
We could have this in the scan state but i'm worried if it'll get too big. I'll create an issue for this